src/modules/core/components/SingleUserPicker/utils.ts

```diff
@@ -7,20 +7,20 @@ export const filterUserSelection = (data, filterValue) => {
     return data;
   }
 
-  const filtered = data.filter(
-    (user) =>
+  const filtered = data.filter((user) => {
+    const { username } = user.profile;
+    return (
       user &&
       user.profile &&
```

Comment:
Aren't you already checking for it's existence above? 

You should just check for `username` in this instance, which should be enough.